diff --git a/web/dist/dev.cjs b/web/dist/dev.cjs
index 5acebb9500f52ce5360dccb2c7c247f42bf72e51..3a24765f8cf034424ce72581003aeecd87d1d4fb 100644
--- a/web/dist/dev.cjs
+++ b/web/dist/dev.cjs
@@ -750,7 +750,7 @@ function Portal(props) {
       insert(renderRoot, content);
       el.appendChild(container);
       props.ref && props.ref(container);
-      solidJs.onCleanup(() => el.removeChild(container));
+      solidJs.onCleanup(() => container.remove());
     }
   }, undefined, {
     render: !hydrating
@@ -769,7 +769,8 @@ function createDynamic(component, props) {
         return solidJs.untrack(() => component(props));
       case "string":
         const isSvg = SVGElements.has(component);
-        const el = solidJs.sharedConfig.context ? getNextElement() : createElement(component, isSvg, solidJs.untrack(() => props.is));
+        const createDynamicElement = () => createElement(component, isSvg, solidJs.untrack(() => props.is));
+        const el = isHydrating() ? getNextElement(createDynamicElement) : createDynamicElement();
         spread(el, props, isSvg);
         return el;
     }
diff --git a/web/dist/dev.js b/web/dist/dev.js
index 5243efc9a04e8817fbd4fb589e64c45896f9ff22..b276b9b55f851a9377b28551ac733c8f81d542d8 100644
--- a/web/dist/dev.js
+++ b/web/dist/dev.js
@@ -749,7 +749,7 @@ function Portal(props) {
       insert(renderRoot, content);
       el.appendChild(container);
       props.ref && props.ref(container);
-      onCleanup(() => el.removeChild(container));
+      onCleanup(() => container.remove());
     }
   }, undefined, {
     render: !hydrating
@@ -768,7 +768,8 @@ function createDynamic(component, props) {
         return untrack(() => component(props));
       case "string":
         const isSvg = SVGElements.has(component);
-        const el = sharedConfig.context ? getNextElement() : createElement(component, isSvg, untrack(() => props.is));
+        const createDynamicElement = () => createElement(component, isSvg, untrack(() => props.is));
+        const el = isHydrating() ? getNextElement(createDynamicElement) : createDynamicElement();
         spread(el, props, isSvg);
         return el;
     }
diff --git a/web/dist/web.cjs b/web/dist/web.cjs
index 2923e9ba92ce5b240f755826012a27184b03e285..6c8095ce0dc7e9cb70f3879aa4ea5cd35b57fc16 100644
--- a/web/dist/web.cjs
+++ b/web/dist/web.cjs
@@ -742,7 +742,7 @@ function Portal(props) {
       insert(renderRoot, content);
       el.appendChild(container);
       props.ref && props.ref(container);
-      solidJs.onCleanup(() => el.removeChild(container));
+      solidJs.onCleanup(() => container.remove());
     }
   }, undefined, {
     render: !hydrating
@@ -758,7 +758,8 @@ function createDynamic(component, props) {
         return solidJs.untrack(() => component(props));
       case "string":
         const isSvg = SVGElements.has(component);
-        const el = solidJs.sharedConfig.context ? getNextElement() : createElement(component, isSvg, solidJs.untrack(() => props.is));
+        const createDynamicElement = () => createElement(component, isSvg, solidJs.untrack(() => props.is));
+        const el = isHydrating() ? getNextElement(createDynamicElement) : createDynamicElement();
         spread(el, props, isSvg);
         return el;
     }
diff --git a/web/dist/web.js b/web/dist/web.js
index dc905bedb850b8319880492532d9b77a83ad2263..39074b06e4c9dfff3f3f7ad48afbc97fee74dd02 100644
--- a/web/dist/web.js
+++ b/web/dist/web.js
@@ -741,7 +741,7 @@ function Portal(props) {
       insert(renderRoot, content);
       el.appendChild(container);
       props.ref && props.ref(container);
-      onCleanup(() => el.removeChild(container));
+      onCleanup(() => container.remove());
     }
   }, undefined, {
     render: !hydrating
@@ -757,7 +757,8 @@ function createDynamic(component, props) {
         return untrack(() => component(props));
       case "string":
         const isSvg = SVGElements.has(component);
-        const el = sharedConfig.context ? getNextElement() : createElement(component, isSvg, untrack(() => props.is));
+        const createDynamicElement = () => createElement(component, isSvg, untrack(() => props.is));
+        const el = isHydrating() ? getNextElement(createDynamicElement) : createDynamicElement();
         spread(el, props, isSvg);
         return el;
     }
